### PR TITLE
Merge maintenance/remove-the-need-of-fields-for-option-properties into develop

### LINKED ISSUE #101: [Core] Remove the need of fields for option properties
Currently each of our option properties needs a separate field which holds it's value. We should remove this dependency and use a Dictionary internally to save the values like as in the BindingBase

### PR DESCRIPTION
**Pull request by LegendaryB**
Closes #101

### CODE REVIEW

File: src/Messaging/Configuration/SubscriptionManagerOptions.cs [hunk 1]
```diff
@@ -6,15 +6,18 @@ namespace CodeMonkeys.Messaging
...
         {
-            get => _flushSubscriptionsPeriod;
-            set => SetValue(ref _flushSubscriptionsPeriod, value);
+            get => GetValue<TimeSpan>();
+            set => SetValue(value);
+        }
+
+        public SubscriptionManagerOptions()
+        {
+            FlushSubscriptionsPeriod = TimeSpan.FromSeconds(10);
```
Comment (Pantheas) [inline src/Messaging/Configuration/SubscriptionManagerOptions.cs:20]: Codacy Issue found: [Assign this magic number '10' to a well-named (variable|constant), and use the (variable|constant) instead.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5600476)

**Comment by Pantheas** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 1
           

Complexity increasing per file
==============================
- src/Core/PropertyBag.cs  1
         

Complexity decreasing per file
==============================
+ src/Core/Configuration/Options.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5600476&bid=18348320)

File: src/MVVM/PropertyChanged/BindingBase/BindingBase.cs [hunk 2]
```diff
@@ -26,8 +26,8 @@ public partial class BindingBase :
 
 
 
-        private readonly ConcurrentDictionary<string, object> _properties =
-            new ConcurrentDictionary<string, object>();
+        private readonly PropertyBag _propertyBag =
+            new PropertyBag();
 
         private readonly ConcurrentDictionary<object, string> _nestedProperties =
```
Comment (Pantheas) [inline src/MVVM/PropertyChanged/BindingBase/BindingBase.cs:32]: Can't this be a `PropertyBag` aswell?
We should check this :)

**Review by Pantheas — CHANGES_REQUESTED** [review]

File: src/Messaging/Configuration/SubscriptionManagerOptions.cs (hunk 1 above)
Comment (LegendaryB) [reply src/Messaging/Configuration/SubscriptionManagerOptions.cs:20]: Implemented in #105

**Review by Pantheas — APPROVED** [review]

**Pull request merged**